#iZ and CWS
o http://www.openoffice.org/issues/show_bug.cgi?id=40180

Description

Vendor name of FreeBSD's Java
(http://www.freebsd.org/cgi/cvsweb.cgi/ports/java/jdk14/)
is Sun Microsystems Inc. so that we can use as Linux version
only difference is the out put of java -version
java version "1.4.2-p6"
trailing -pXX is different.

Index: jvmfwk/plugins/sunmajor/pluginlib/sunversion.cxx
===================================================================
RCS file: /cvs/udk/jvmfwk/plugins/sunmajor/pluginlib/sunversion.cxx,v
retrieving revision 1.5
diff -u -r1.5 sunversion.cxx
--- jvmfwk/plugins/sunmajor/pluginlib/sunversion.cxx	16 Dec 2004 11:45:15 -0000	1.5
+++ jvmfwk/plugins/sunmajor/pluginlib/sunversion.cxx	30 Dec 2004 12:55:14 -0000
@@ -231,6 +231,8 @@
         m_preRelease = getPreRelease(pCur);
         if (m_preRelease == Rel_NONE)
             return false;
+        if (m_preRelease == Rel_FreeBSD)
+            return true;
     }
     else
     {
@@ -267,6 +269,20 @@
         return Rel_RC2;
     else if (! strcmp(szRelease, "rc3"))
         return Rel_RC3;
+    else if (! strcmp(szRelease, "p5"))
+        return Rel_FreeBSD;
+    else if (! strcmp(szRelease, "p6"))
+        return Rel_FreeBSD;
+    else if (! strcmp(szRelease, "p7"))
+        return Rel_FreeBSD;
+    else if (! strcmp(szRelease, "p8"))
+        return Rel_FreeBSD;
+    else if (! strcmp(szRelease, "p9"))
+        return Rel_FreeBSD;
+    else if (! strcmp(szRelease, "p10"))
+        return Rel_FreeBSD;
+    else if (! strcmp(szRelease, "p11"))
+        return Rel_FreeBSD;
     else
         return Rel_NONE;
 }
Index: jvmfwk/plugins/sunmajor/pluginlib/sunversion.hxx
===================================================================
RCS file: /cvs/udk/jvmfwk/plugins/sunmajor/pluginlib/sunversion.hxx,v
retrieving revision 1.5
diff -u -r1.5 sunversion.hxx
--- jvmfwk/plugins/sunmajor/pluginlib/sunversion.hxx	16 Dec 2004 11:45:27 -0000	1.5
+++ jvmfwk/plugins/sunmajor/pluginlib/sunversion.hxx	30 Dec 2004 12:55:14 -0000
@@ -107,6 +107,10 @@
         Rel_RC1,
         Rel_RC2,
         Rel_RC3
+#if defined(FREEBSD)
+	,
+        Rel_FreeBSD
+#endif
     };
 
     //contains major,minor,micro,update
